fs/mmap/fs_mmap.c

```diff
@@ -52,6 +52,10 @@ static int file_mmap_(FAR struct file *filep, FAR void *start,
   FAR void *addr;
   int ret;
 
+#ifdef CONFIG_MM_VM_MAP
```

Comment:
file backup's mmap may need this list too, which isn't configurable now.

Reply:
sorry can you elaborate more? There is actually no implementation of mmapping a "real" file from filesystems atm.

Reply:
> sorry can you elaborate more? There is actually no implementation of mmapping a "real" file from filesystems atm.

There is at least two implementation(ROMFS and TMPFS) which utilize FIOC_MMAP:
https://github.com/apache/incubator-nuttx/blob/master/fs/mmap/fs_mmap.c#L144
Since I saw you add FIOC_UMMAP, it most likely will be used in munmap, and then the list.

Reply:
Those filesystems don't impelment unmap, probably for the reason that this kind of vm mapping was not available before, so it was not possible to go from the virtual address to the driver who mmapped it... Note that I just added the FIOC_UNMAP, since I wanted to implement my shm thingy fully (i.e. also support unmap).

Reply:
looking at tmpfs more closely; it frees the memory when unlinking the file. It just doesn't support first mmap and then unlink.

Reply:
as far as I can see, it now works so that if you have CONFIG_VM_MAP enabled, the both romfs and tmpfs work the same as before, with the exception that the mappings are pushed into vm_maps, and removed from there at munmap.

If you don't have the CONFIG_VM_MAP enabled, both romfs and tmpfs work the same as before.

So where is the problem you are seeing?